octodns/provider/mythicbeasts.py

```diff
@@ -0,0 +1,420 @@
+    _data_for_TXT = _data_for_multiple
+    _data_for_A = _data_for_multiple
+    _data_for_AAAA = _data_for_multiple
+
+    def populate(self, zone, target=False, lenient=False):
+        self.log.debug('populate: name=%s, target=%s, lenient=%s', zone.name,
+                       target, lenient)
+
+        resp = self.records(zone.name)
+
+        before = len(zone.records)
+        exists = False
+        data = dict()
+
+        exists = True
+        for line in resp.content.splitlines():
+            match = re.match(
+                '^(\\S+)\\s+(\\d+)\\s+(\\S+)\\s+(.*)$',
+                line,
+                re.IGNORECASE)
+
+            if match is None:
+                self.log.debug('failed to match line: %s', line)
+                continue
+
+            if match.group(1) == '@':
+                _name = ''
+            else:
+                _name = match.group(1)
+
+            _type = match.group(3)
+            _ttl = int(match.group(2))
+            _value = match.group(4).strip()
+
+            if _type == 'TXT':
+                _value = _value.replace(';', '\\;')
+
+            if hasattr(self, '_data_for_{}'.format(_type)):
+
+                if _type not in data:
+                    data[_type] = dict()
+
+                if _name not in data[_type]:
+                    data[_type][_name] = {
+                        'raw_values': [{'value': _value, 'ttl': _ttl}],
+                        'name': _name,
+                        'zone': zone.name,
+                    }
+
+                else:
+                    data[_type][_name].get('raw_values').append(
+                        {'value': _value, 'ttl': _ttl}
+                    )
+            else:
+                self.log.debug('skipping %s as not supported', _type)
+
+        for _type in data:
+            for _name in data[_type]:
+                data_for = getattr(self, '_data_for_{}'.format(_type))
+
+                record = Record.new(
+                    zone,
+                    _name,
+                    data_for(_type, data[_type][_name]),
+                    source=self
+                )
+                zone.add_record(record, lenient=lenient)
+
+        self.log.debug('populate:   found %s records, exists=%s',
+                       len(zone.records) - before, exists)
+
+        return exists
+
+    def _compile_commands(self, action, change):
+        commands = []
+
+        record = None
+
+        if action == 'ADD':
+            record = change.new
+        else:
+            record = change.existing
+
+        hostname = remove_trailing_dot(record.fqdn)
+        ttl = record.ttl
+        _type = record._type
+
+        if _type == 'ALIAS':
+            _type = 'ANAME'
+
+        if hasattr(record, 'values'):
+            values = record.values
+        else:
+            values = [record.value]
+
+        base = '{} {} {} {}'.format(action, hostname, ttl, _type)
+
+        if re.match('[A]{1,4}', _type) is not None:
```

Comment:
If i understand this I think it'd be clearer as:

```suggestion
        if _type in ('A', 'AAAA'):
```